src/ripple/app/tx/Transaction.cpp

```diff
@@ -52,6 +52,30 @@ Transaction::Transaction (STTx::ref sit, Validate validate)
     }
 }
 
+Transaction::Transaction (STTx::ref sit, Validate validate, std::string& reason)
+    : mInLedger (0),
+      mStatus (INVALID),
+      mResult (temUNCERTAIN),
+      mTransaction (sit)
+{
+    try
+    {
+        mFromPubKey.setAccountPublic (mTransaction->getSigningPubKey ());
+        mTransactionID  = mTransaction->getTransactionID ();
+        mAccountFrom    = mTransaction->getSourceAccount ();
+    }
+    catch (...)
+    {
+        return;
```

Comment:
Hi Mickey, I really like adding the reason for the construction failure.

I'm wondering if there's anything reasonable you could write into `reason` here in this catch.

Do you know if the other constructor is still needed?  If not it would be great to remove it.  If it is still needed, you might be able to delegate that constructor to this one (by passing a temporary `std::string` in the delegation).  That would reduce code duplication just a bit.